timemachine/fe/single_topology_v3.py

```diff
@@ -735,3 +742,182 @@ def setup_intermediate_state(self, lamb):
         )
 
         return system.VacuumSystem(bond, angle, torsion, nonbonded, chiral_atom, chiral_bond)
+
+    def combine_with_host(
+        self,
+        host_system,
+        lamb,
+    ):
+        """
+        Setup host guest system. Bonds, angles, torsions, chiral_atom, chiral_bond and nonbonded terms are
+        combined. In particular:
+
+        1) Bond, angle, torsion, chiral_atom, chiral_bond simply have the idxs shifted by num_host_atoms.
+        2) Host-host and host-guest interactions use a nonbonded potential, with exclusions set to the
+            original host-host exclusions, in addition to *all* guest-guest interactions being excluded.
+            Host-guest interactions are implemented as follows:
+            i) at lambda = 0, the dummy atoms of mol_a are fully interacting, and become non-interacting at lambda = 1
+            ii) at lambda = 0, the dummy atoms of mol_b are non-interacting, and become fully interacting at lambda = 1
+            iii) the core atoms have parameters interpolated from mol_a's qlj to mol_b's qlj.
+        3) guest-guest interactions use pre-computed nonbonded interactions implemented as a pairlist.
```

Comment:
Initially I didn't follow how this function implements 3) -- does this happen elsewhere?